editor/components/post-permalink/index.js

```diff
@@ -14,55 +14,151 @@ import { Dashicon, ClipboardButton, Button } from '@wordpress/components';
  * Internal Dependencies
  */
 import './style.scss';
-import { isEditedPostNew, getEditedPostAttribute } from '../../store/selectors';
+import { isEditedPostNew, getEditedPostAttribute, getEditedPostPreviewLink } from '../../store/selectors';
+import { editPost } from '../../store/actions';
+import { getWPAdminURL } from '../../utils/url';
 
 class PostPermalink extends Component {
 	constructor() {
 		super( ...arguments );
+
 		this.state = {
-			showCopyConfirmation: false,
+			iconClass: '',
+			samplePermalink: '',
+			samplePermalinkPrefix: '',
+			samplePermalinkSuffix: '',
+			postName: '',
+			editedPostName: '',
+			editingPermalink: false,
 		};
-		this.onCopy = this.onCopy.bind( this );
-		this.onFinishCopy = this.onFinishCopy.bind( this );
+
+		this.onSavePermalink = this.onSavePermalink.bind( this );
 	}
 
-	componentWillUnmount() {
-		clearTimeout( this.dismissCopyConfirmation );
+	componentWillMount() {
+		if ( this.props.samplePermalinkData ) {
+			this.updateSamplePermalink( this.props.samplePermalinkData[ 0 ], this.props.samplePermalinkData[ 1 ] );
+		}
 	}
 
-	onCopy() {
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.samplePermalinkData !== nextProps.samplePermalinkData ) {
+			this.updateSamplePermalink( nextProps.samplePermalinkData[ 0 ], nextProps.samplePermalinkData[ 1 ] );
+		}
+	}
+
+	/**
+	 * Updates the samplePermalink data in the state.
+	 *
+	 * @param {string} template Permalink structure to replace the tags in.
+	 * @param {string} postName The post slug to use in replacements.
+	 */
+	updateSamplePermalink( template, postName ) {
+		const regex = /%(?:postname|pagename)%/;
```

Comment:
If this is constant, seems we could move it to top-of-file (shared scope) and document appropriately for what it's matching.